litestar/_openapi/schema_generation/schema.py

```diff
@@ -549,14 +558,99 @@ def create_schema_for_pydantic_model(
         },
         type=OpenAPIType.OBJECT,
         title=_get_type_schema_name(field_type),
-        description=getdoc(field_type) or None,
+    )
+
+
+def create_schema_for_attrs_class(
+    field_type: type[AttrsInstance],
+    generate_examples: bool,
+    plugins: list[OpenAPISchemaPluginProtocol],
+    schemas: dict[str, Schema],
+) -> Schema:
+    """Create a schema object for a given attrs class.
+
+    Args:
+        field_type: An attrs class.
+        generate_examples: Whether to generate examples if none are given.
+        plugins: A list of plugins.
+        schemas: A mapping of namespaces to schemas - this mapping is used in the OA components section.
+
+    Returns:
+        A schema instance.
+    """
+    from attr import NOTHING
+    from attrs import fields_dict
+
+    field_type_hints = get_type_hints(field_type)
+    return Schema(
+        required=sorted(
+            [
+                field_name
+                for field_name, attribute in fields_dict(field_type).items()
+                if attribute.default is NOTHING and not is_optional_union(field_type_hints[field_name])
+            ]
+        ),
+        properties={
+            k: create_schema(
+                field=SignatureField.create(field_type=v, name=k),
+                generate_examples=generate_examples,
+                plugins=plugins,
+                schemas=schemas,
+            )
+            for k, v in field_type_hints.items()
+        },
+        type=OpenAPIType.OBJECT,
+        title=_get_type_schema_name(field_type),
+    )
+
+
+def create_schema_for_struct_class(
+    field_type: type[Struct],
+    generate_examples: bool,
+    plugins: list[OpenAPISchemaPluginProtocol],
+    schemas: dict[str, Schema],
+) -> Schema:
+    """Create a schema object for a given msgspec.Struct class.
+
+    Args:
+        field_type: A msgspec.Struct class.
+        generate_examples: Whether to generate examples if none are given.
+        plugins: A list of plugins.
+        schemas: A mapping of namespaces to schemas - this mapping is used in the OA components section.
+
+    Returns:
+        A schema instance.
+    """
+
+    field_type_hints = get_type_hints(field_type)
```

Comment:
Instead of using `get_type_hints` here, you could rely on the `.type` attribute of the output of `msgspec_struct_fields` below. This will provide the same information - no sense in retrieving it twice. https://jcristharif.com/msgspec/api.html#msgspec.structs.FieldInfo

Reply:
done